engine/rest-client/src/main/java/org/datacleaner/restclient/ComponentList.java

```diff
@@ -50,11 +51,13 @@ public void add(ComponentInfo componentInfo) {
      */
     @JsonInclude(JsonInclude.Include.NON_NULL)
     public static class ComponentInfo {
-
         private String name = "";
         private String description = "";
         private String createURL = "";
+        private Set<String> categoryNames = new HashSet<>();
+        private String superCategoryName = null;
         private Map<String, PropertyInfo> properties = new HashMap<>();
+        private byte[] iconData = null;
```

Comment:
It will serialize as base64 string. If we look on the actual use-case: Desktop queries list of components and always wants an icon for each component. Doing it in one query is more effective than downloading icons one by one after the component list.
BTW, If somebody don't need icons, there is a boolean parameter to the component-list query which will prevent the icons data in the response (default is false).